prime-router/src/main/kotlin/fhirengine/translation/hl7/schema/ConfigSchemaReader.kt

```diff
@@ -70,10 +75,13 @@ object ConfigSchemaReader : Logging {
             throw SchemaException(msg, e)
         }
 
+        if (ancestry.contains(rawSchema.name)) throw HL7ConversionException("Circular reference detected in schema")
+        rawSchema.ancestry = ancestry + rawSchema.name!!
```

Comment:
I think you can just move this down to line 84 without using rawSchema.ancestry.  I do not think there is a need for rawSchema.ancestry at that point.

Reply:
Gave that a shot, but it doesn't work without having the object reference unless plenty more is done